static/srcView.js

```diff
@@ -16,30 +16,25 @@ import { MenuHost, Menu } from './menus';
 
 
 // Menus, highlighting on mouseover.
-function add_ref_functionality(self) {
+function add_ref_functionality() {
     for (const el of $("#div_src_view").find(".class_id")) {
         const element = $(el);
         const classes = el.className.split(' ');
         // TODO[ES6]: use classes.find() and then execute following code
-        for (const c of classes) {
-            if (c.startsWith('class_id_')) {
-                element.hover(function() {
-                    $("." + c).css("background-color", "#d5f3b5");
-                }, function() {
-                    $("." + c).css("background-color", "");
-                });
-
-                const id = c.slice('class_id_'.length);
-                const showRefMenu = (ev) => {
-                    self.setState({ refMenu: { "top": ev.pageY, "left": ev.pageX, target: ev.target, id }});
-                    ev.preventDefault();
-                };
-                element.on("contextmenu", showRefMenu);
-                element.addClass("hand_cursor");
-
-                break;
-            }
-        }
+        let c = classes.find((c) => c.startsWith('class_id_'));
+        element.hover(function() {
+            $("." + c).css("background-color", "#d5f3b5");
+        }, function() {
+            $("." + c).css("background-color", "");
+        });
+
+        const id = c.slice('class_id_'.length);
+        const showRefMenu = (ev) => {
+            this.setState({ refMenu: { "top": ev.pageY, "left": ev.pageX, target: ev.target, id }});
```

Comment:
Hmm, so I'm not confident enough about JS and `this` to be sure that this is wrong - but it seems a little sketchy - are we relying on `this` to still be set by the caller? IIRC this may or may not work depending on strict mode. In any case it seems bad compared to passing an explicit argument for this.